mesheryctl/pkg/utils/platform.go

```diff
@@ -453,6 +453,49 @@ func ChangeManifestVersion(fileName string, version string, filePath string) err
 	return nil
 }
 
+// ChangeManifestServiceType changes the service type of meshery deployment
+func ChangeManifestServiceType(fileName string, serviceType string, filePath string) error {
+	// setting up config type to yaml files
+	ViperK8s.SetConfigType("yaml")
+
+	// setting up config file
+	ViperK8s.SetConfigFile(filePath)
+	err := ViperK8s.ReadInConfig()
+	if err != nil {
+		return fmt.Errorf("unable to read config %s | %s", fileName, err)
+	}
+
+	compose := K8sCompose{}
+	yamlFile, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return err
+	}
+	// unmarshal the file into structs
+	err = yaml.Unmarshal(yamlFile, &compose)
+	if err != nil {
+		return fmt.Errorf("unable to unmarshal config %s | %s", fileName, err)
+	}
+
+	compose.Spec.Type = serviceType
+
+	ViperK8s.Set("apiVersion", compose.APIVersion)
+	ViperK8s.Set("kind", compose.Kind)
+	ViperK8s.Set("metadata", compose.Metadata)
+	ViperK8s.Set("spec", compose.Spec)
+
+	// Marshal the structs
+	newConfig, err := yaml.Marshal(compose)
+	if err != nil {
+		return fmt.Errorf("unable to marshal config %s | %s", fileName, err)
+	}
+	err = ioutil.WriteFile(filePath, newConfig, 0644)
```

Comment:
ViperK8s seems to store values for config.yaml and using the same here making it store apiVersion, kind, etc.. and when marshalling this data is also added to config.yaml. Should I create another viper instance to store these or use vipercompose? @navendu-pottekkat

Reply:
This was the reason for overwriting @metonymic-smokey

Reply:
@vineethvanga18 We should have different viper instances for different config files. But I think you are already doing that

Reply:
No. ViperK8s is being used for config.yaml. I will create another instance for this

Reply:
Oh, yes, I see it now-

![Screenshot from 2021-05-26 16-51-19](https://user-images.githubusercontent.com/49474499/119651436-a462f600-be42-11eb-81a4-dca35d7af8ee.png)

It is being used to change the endpoints in config.yaml